packages/erk-shared/src/erk_shared/issue_workflow.py

```diff
@@ -0,0 +1,185 @@
+
+
+def validate_issue_for_worktree(
+    issue_info: IssueInfo,
+    *,
+    warn_non_open: bool = True,
+) -> list[str]:
+    """Validate issue is suitable for worktree creation.
+
+    Checks that the issue has the required 'erk-plan' label and
+    optionally warns about non-OPEN issues.
+
+    Args:
+        issue_info: Issue information from GitHub
+        warn_non_open: Whether to generate warning for non-OPEN issues
+
+    Returns:
+        List of warning messages (empty if no warnings)
+
+    Raises:
+        IssueValidationError: If erk-plan label is missing
+    """
+    warnings: list[str] = []
+
+    if "erk-plan" not in issue_info.labels:
+        raise IssueValidationError(
+            f"Issue #{issue_info.number} must have 'erk-plan' label.\n"
+            f"To add the label:\n"
+            f"  gh issue edit {issue_info.number} --add-label erk-plan"
+        )
+
+    if warn_non_open and issue_info.state != "OPEN":
+        warnings.append(f"Issue #{issue_info.number} is {issue_info.state}. Proceeding anyway...")
+
+    return warnings
+
+
+def validate_plan_for_worktree(
+    plan: Plan,
+    *,
+    warn_non_open: bool = True,
+) -> list[str]:
+    """Validate plan is suitable for worktree creation.
+
+    This is the Plan-based variant of validate_issue_for_worktree,
+    used when working with the plan_store abstraction.
+
+    Checks that the plan has the required 'erk-plan' label and
+    optionally warns about non-OPEN plans.
+
+    Args:
+        plan: Plan from plan_store
+        warn_non_open: Whether to generate warning for non-OPEN plans
+
+    Returns:
+        List of warning messages (empty if no warnings)
+
+    Raises:
+        IssueValidationError: If erk-plan label is missing
+    """
+    warnings: list[str] = []
+
+    if "erk-plan" not in plan.labels:
+        raise IssueValidationError(
+            f"Issue #{plan.plan_identifier} must have 'erk-plan' label.\n"
+            f"To add the label:\n"
+            f"  gh issue edit {plan.plan_identifier} --add-label erk-plan"
+        )
+
+    if warn_non_open and plan.state != PlanState.OPEN:
+        warnings.append(
+            f"Issue #{plan.plan_identifier} is {plan.state.value}. Proceeding anyway..."
+        )
+
+    return warnings
+
+
+def prepare_plan_for_worktree(
+    plan: Plan,
+    timestamp: datetime,
+) -> IssueBranchSetup:
+    """Prepare plan data for worktree creation.
+
+    This is the Plan-based variant of prepare_issue_for_worktree,
+    used when working with the plan_store abstraction.
+
+    Generates branch name and worktree name from plan metadata.
+    Does NOT create the branch or worktree - just computes names.
+
+    Args:
+        plan: Plan from plan_store
+        timestamp: Timestamp for branch name suffix
+
+    Returns:
+        IssueBranchSetup with computed names and plan data
+    """
+    # Plan uses plan_identifier (string) but we need int for issue number
+    issue_number = int(plan.plan_identifier)
```

Comment:
we should explicitly check to see if it can bo converted to an int and give a helpful error message